dask/array/tests/test_array_core.py

```diff
@@ -1959,13 +1959,22 @@ def test_store_compute_false():
 
     at = np.zeros(shape=(4, 4))
     bt = np.zeros(shape=(4, 4))
-
     dat, dbt = store([a, b], [at, bt], compute=False, return_stored=True)
     assert isinstance(dat, Array) and isinstance(dbt, Array)
     assert (at == 0).all() and (bt == 0).all()
     assert (dat.compute() == at).all() and (dbt.compute() == bt).all()
     assert (at == 2).all() and (bt == 3).all()
 
+    at = np.zeros(shape=(4, 4))
+    bt = np.zeros(shape=(4, 4))
+    dat, dbt = store(
+        [a, b], [at, bt], compute=False, return_stored=True, load_stored=False
+    )
+    assert isinstance(dat, Array) and isinstance(dbt, Array)
+    assert (at == 0).all() and (bt == 0).all()
+    dask.compute(dat, dbt)
```

Comment:
Note this will return an (8,8) array not (4,4) for `dat` and `bat` since each chunk returns a (4,4) array.

https://github.com/dask/dask/blob/fa8fecf10a94971f2f31df57d504d25bef4dd57e/dask/array/core.py#L4550-L4555

This is ugly, but totally backwards compatible, and I don't expect anyone will use it other than icechunk :).

I use this like-so in icechunk, where I extract the Zarr store from the Zarr array that is returned in each chunk.

https://github.com/earth-mover/icechunk/blob/dcdfa82e23427facebc921919c27dd615f11db72/icechunk-python/python/icechunk/dask.py#L31